src/app/clusters/diagnostic-logs-server/BDXDiagnosticLogsProvider.cpp

```diff
@@ -191,7 +191,11 @@ void BDXDiagnosticLogsProvider::OnAckReceived()
     // If the buffer has empty space, end the log collection session.
     if (isEndOfLog)
     {
-        mDelegate->EndLogCollection(mLogSessionHandle);
+        err = mDelegate->EndLogCollection(mLogSessionHandle, err);
+        if (err == CHIP_ERROR_NOT_IMPLEMENTED)
+        {
+            ChipLogError(DeviceLayer, "EndLogCollection not implemented in the delegate. Falling back to default behavior.");
```

Comment:
Which default behavior?  We do exactly the same thing no matter what EndLogCollection returns, raising questions about why it returns anything at all.....

Reply:
@bzbarsky-apple I added this to notify the application side if EndLogCollection is not implemented, in case the user misses it, as we have provided default implementations for earlier virtual methods.
I will remove it.